### ISSUE TITLE
Invert cutting direction

### ISSUE THREAD

**Issue by newhinton** [Timmmm/robocut#4]
Hello! I have searched for a tool that can replace the software for a silouette portrait. Yours seems fitting! 

However, while trying it out it started cutting "from the bottom". I have no fitting sheets, so robocut "pulls in" a full sheet, and therefore dispensing the sheet altogether. When now trying to cut, the vinyl is missing. 

Is it possible to invert the cutting direction, eg. inserting the sheet and starting to cut in the top left corner instead of bottom right?

**Comment by Timmmm** [comment]
Yeah I could definitely add that. The coordinate system is a bit of a mess!
Though, maybe I've misunderstood but couldn't you just edit your SVG in
Inkscape to move the content to the other end of the page?

Could you post an example SVG?

**Comment by newhinton** [comment]
![test](https://user-images.githubusercontent.com/25279821/163689674-c38dacce-3709-4fc3-8ba2-bb5bbf307cf7.jpg)


I have created this ...wonderful... drawing!

Legend: 
Red: Cutting Area
Orange line without arrow: Axis on which the knive moves left to right
Blue thick line: Vinyl-feed-roller
Grey rectangle: paper/vinyl
Turquoise line: Items to be cut. The dot is the first item to be cut, the arrow is the direction in which items should be processed (roughly)

Green Arrow: Movement of vinyl BEFORE cutting starts
Yellow Arrow: Movement of vinyl during cutting. 

Left side: Observed behaviour of robocut. The vinyl is fully moved through the cutter. If it is shorter than the paper expected by robocut, it looses the connection and the cutter starts moving it around so that it cannot be cut or moved anymore. Its ejection direction is back towards the user.

Right side: How Silouhette Studio does it. There is no movement of the vinyl until the cutting starts. When it starts cutting, it moves the vinyl into the cutter, until it runs out. The benefit is that we dont have to supply a full A4 sheet, because we dont have to pull a full sheet through the printer before cutting can begin. (In reality, after Silouhette Studio is done, it also moves the paper back the same amount it moved it in, so that it pops out, but thats just a nice to have as it can be done with the buttons on the device)

I hope this helps!

**Comment by Timmmm** [comment]
Ah right I see what you mean. I think this is just because it cuts in SVG order currently. Fortunately I have already written code to cut in a more sensible order. Btw this is basically the Travelling Salesman Problem! Actually it's called the Travelling Salesman Problem with Neighbourhoods (since the things you visit are regions not points).

Anyway I wrote some code a couple of years ago to try and do shortest path sort of stuff. Also added a picture of the vinyl cutter to make the orientation a bit more obvious.

![image](https://user-images.githubusercontent.com/376842/163719254-307201c6-4618-4b8b-84c2-b64e2420bfd1.png)

I am trying to finally get a new release out soon so I will try to make sure this works as you describe.

**Comment by mikhailmihalkov** [comment]
https://github.com/Jeinzi/SMTCut
Timmmm Tell me where can there be an error in the code? The problem is that no matter how many degrees I turn gerber file in gerbv (180/90/270) Portrait 3 stubbornly cuts from 90 degrees angles relative to the fiducial. Where in the code is this parameter?  If you look at the resulting pdf, there it flips in the desired direction but the plotter cuts with a clockwise rotation angle of 90 degrees
